browser/scripts/ui.js

```diff
@@ -196,20 +260,80 @@ VizorUI.prototype.onSearchResultsChange = function() {
 		presetsLib.height('auto');
 		presetsList.height(maxHeight);
 	}
-	this.updateState();
 };
 
 VizorUI.prototype.onBtnPresetsClicked = function() {
 	if (!this.isVisible()) return false;
-	this.state.visibility.panel_presets = !this.state.visibility.panel_presets;
-	this.applyVisibility();
+	var v = this.state.visibility;
+	if (!v.floating_panels) {
+		// nothing displayed so we expect just the presets
+		v._.panel_chat = false
+		v._.panel_assets = false;
+		v._.panel_presets = true;
+		v.floating_panels = true;
+	} else {
+		v.panel_presets = !v.panel_presets;
+	}
 	return false;
 }
 
 VizorUI.prototype.onBtnAssetsClicked = function() {
 	if (!this.isVisible()) return false;
-	this.state.visibility.panel_assets = !this.state.visibility.panel_assets;
-	this.applyVisibility();
+	var v = this.state.visibility;
+	if (!v.floating_panels) {
+		// nothing displayed so we expect just the assets
+		v._.panel_chat = false
```

Comment:
i can only move this complexity about, I don't think I can avoid it altogether, this is an interaction issue and I've tried to cover an inherently messy model.